src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1,394 +1,625 @@
+            LOGGER.debug("UpdateRecommendations API request count: {} failed", calCount);
+            kruizeObject.setValidation_data(new ValidationOutputData(false, validationMessage, HttpServletResponse.SC_BAD_REQUEST));
+            return kruizeObject;
         }
-        RecommendationConfigItem recommendationConfigItem = null;
-        String format = "";
-        List<Double> cpuUsageList = filteredResultsMap.values()
-                .stream()
-                .map(e -> {
-                    Optional<MetricResults> cpuUsageResults = Optional.ofNullable(e.getMetricResultsMap().get(AnalyzerConstants.MetricName.cpuUsage));
-                    Optional<MetricResults> cpuThrottleResults = Optional.ofNullable(e.getMetricResultsMap().get(AnalyzerConstants.MetricName.cpuThrottle));
-                    double cpuUsageAvg = cpuUsageResults.map(m -> m.getAggregationInfoResult().getAvg()).orElse(0.0);
-                    double cpuUsageMax = cpuUsageResults.map(m -> m.getAggregationInfoResult().getMax()).orElse(0.0);
-                    double cpuUsageSum = cpuUsageResults.map(m -> m.getAggregationInfoResult().getSum()).orElse(0.0);
-                    double cpuThrottleAvg = cpuThrottleResults.map(m -> m.getAggregationInfoResult().getAvg()).orElse(0.0);
-                    double cpuThrottleMax = cpuThrottleResults.map(m -> m.getAggregationInfoResult().getMax()).orElse(0.0);
-                    double cpuThrottleSum = cpuThrottleResults.map(m -> m.getAggregationInfoResult().getSum()).orElse(0.0);
-                    double cpuRequestInterval = 0.0;
-                    double cpuUsagePod = 0;
-                    int numPods = 0;
-
-                    // Use the Max value when available, if not use the Avg
-                    double cpuUsage = (cpuUsageMax > 0) ? cpuUsageMax : cpuUsageAvg;
-                    double cpuThrottle = (cpuThrottleMax > 0) ? cpuThrottleMax : cpuThrottleAvg;
-                    double cpuUsageTotal = cpuUsage + cpuThrottle;
-
-                    // Usage is less than 1 core, set it to the observed value.
-                    if (CPU_ONE_CORE > cpuUsageTotal) {
-                        cpuRequestInterval = cpuUsageTotal;
+        interval_end_time = Utils.DateUtils.getTimeStampFrom(KruizeConstants.DateFormats.STANDARD_JSON_DATE_FORMAT,
+                intervalEndTimeStr);
+        LOGGER.debug("UpdateRecommendations API request count: {} experiment_name : {} and interval_start_time : {} and interval_end_time : {} ", calCount, experiment_name, intervalStartTimeStr, intervalEndTimeStr);
+        LOGGER.debug("experiment_name : {} and interval_start_time : {} and interval_end_time : {} ", experiment_name, intervalStartTimeStr, intervalEndTimeStr);
+
+        Map<String, KruizeObject> mainKruizeExperimentMAP = new ConcurrentHashMap<>();
+        Map<String, Terms> terms = new HashMap<>();
+        try {
+            new ExperimentDBService().loadExperimentFromDBByName(mainKruizeExperimentMAP, experiment_name);
```

Comment:
abstract out for getting data
from DataSource identify where to fetch data  is it via DB or RestAPI

Reply:
Ok getting KruizeObject is always from DB not via restAPI
But before fetching results always check for is it via DB or via restAPI

Reply:
Done